zoe-core/src/functions/admin.kt

```diff
@@ -362,15 +400,41 @@ data class DeploySchemaConfig(
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
 @JsonSubTypes(
-    JsonSubTypes.Type(value = ActualRunResponse::class, name = "actual"),
-    JsonSubTypes.Type(value = DryRunResponse::class, name = "dry")
+    JsonSubTypes.Type(value = DeploySchemaResponse.ActualRunResponse::class, name = "actual"),
+    JsonSubTypes.Type(value = DeploySchemaResponse.DryRunResponse::class, name = "dry")
 )
 sealed class DeploySchemaResponse {
     data class ActualRunResponse(val id: Int, val subject: String) : DeploySchemaResponse()
     data class DryRunResponse(val subject: String, val schema: String, val registry: String) :
         DeploySchemaResponse()
 }
 
+data class DeleteSchemaConfig(
+    val registry: String,
+    val props: Map<String, String?>,
+    val subject: String,
+    val schemaVersion: Int?,
+    val hardDelete: Boolean,
```

Comment:
If I understand correctly, this `hardDelete` corresponds to `permanent = true` in the schema registry delete request. In that case, shouldn't we stick with `permanent` here as the name of the attribute and also the name of the option in the zoe CLI to stick closer to the schema registry API? `zoe schemas delete --permanent` instead of `--hard`? What do you think?

Reply:
I used `hardDelete` as an effort to follow Confluent's terminology on this: [Soft](https://docs.confluent.io/platform/current/schema-registry/schema-deletion-guidelines.html#soft-delete-a-schema) & [Hard](https://docs.confluent.io/platform/current/schema-registry/schema-deletion-guidelines.html#hard-delete-a-schema) deletes (i.e metadata including schema IDs).
That being said, you're right and the actual request parameter is called `permanent`. I'm not super opinionated on this so happy to change if you think it makes more sense!

Reply:
I wasn't aware of that terminology being on confluent's docs. In that case it's perfectly fine 👍🏼 . Let's keep the Soft vs Hard wording